lib/python/picongpu/picmi/grid.py

```diff
@@ -59,26 +59,35 @@ def get_as_pypicongpu(self):
             "periodic": grid.BoundaryCondition.PERIODIC,
         }
 
-        assert self.bc_xmin in picongpu_boundary_condition_by_picmi_id, \
+        assert self.lower_boundary_conditions[0] in \
+            picongpu_boundary_condition_by_picmi_id, \
             "X: boundary condition not supported"
-        assert self.bc_ymin in picongpu_boundary_condition_by_picmi_id, \
+        assert self.lower_boundary_conditions[1] in \
+            picongpu_boundary_condition_by_picmi_id, \
             "Y: boundary condition not supported"
-        assert self.bc_zmin in picongpu_boundary_condition_by_picmi_id, \
+        assert self.lower_boundary_conditions[2] in \
+            picongpu_boundary_condition_by_picmi_id, \
```

Comment:
could this be solved by:
```python
for tmp_boundar_condition in self.lower_boundary_conditions:
    assert tmp_boundar_condition in picongpu_boundary_condition_by_picmi_id
```

Reply:
In principle yes, but we want to specify the specific boundary condition the user misconfigured, to make it more user friendly.

I would argue good enough.